### ISSUE TITLE
Change the editor gizmos for 2D nodes when they have no "rectangle" 

### ISSUE THREAD

**Issue by groud** [godotengine/godot#17101]
**Godot version:**
3.1

This is a feature request.
We could replace the default placeholder rect used to select nodes in the 2D editor by a "point" (instead of a rectangle). This would avoid missclicks when trying to move a node and scale it instead (which is quite unusual). Node that have a meaninful rectangle, such as Sprites, can keep the rectangle.

Also, to identify more clearly what are each node, we might use each Node's icon to identify its type. Probably with an option to hide them too.

**Comment by Zylann** [comment]
Wouldnt this be annoying if an instanced scene has a Node2D root? Would it make it non-selectable or hardly-selectable due to being a point? (it should at least have a rect encompassing the contents of the scene imo)

**Comment by groud** [comment]
Yes, we can use an encompassing rect for instanced scenes (or "non-selectable children" nodes too).
